depthai_nodes/ml/parsers/parser.py

```diff
@@ -13,78 +27,101 @@ def __init__(self):
 
         self.head_config: Dict = {}
 
-    def build(self, heads: Union[List, Dict], head_name: str = ""):
-        """Initial build method for all parsers. The method sets the head configuration
-        for the parser.
+    def build(self, nn_archive: dai.NNArchive, head_name: str = None):
```

Comment:
Let's go with `head_index: int = None` for now. This is optional if we can't automatically match it - in that case we use the list index provided. But if it is specified we don't do any automatic matching and we use that directly. 
If `parser` field in the head from the provided index doesn't match the parser we want to create we also error out so the user knows that either index or parser specified aren't correct